include/execution.hpp

```diff
@@ -420,8 +424,9 @@ namespace _P2300::execution {
 
   template <class _Sender, class _Env = no_env>
     concept sender =
-      __sender<_Sender, _Env> &&
-      __sender<_Sender, no_env> &&
+      ((__sender<_Sender, _Env> &&
+      __sender<_Sender, no_env> ) ||
+      __awaitable<_Sender,__env_or_void<_Env>>) &&
```

Comment:
I think we want awaitables to satisfy the `__sender` concept checks here. We shouldn't need an extra test for awaitables if `get_completion_signatures` is doing the right thing.

Reply:
Because `__sender` concepts evaluates the `completion_signatures` using the following 
https://github.com/brycelelbach/wg21_p2300_std_execution/blob/2aaf6eb131a1d04c0b3e0dc00f2ce3b960477da1/include/execution.hpp#L298-L304
We can see that for all the awaitables `no_env` and a specific environment would give different completion_signatures`